### PR TITLE
A new version of the portscan alert, with a whitelisting language

### PR DESCRIPTION
**Pull request by mpurzynski**

### CODE REVIEW

**Comment by Phrozyn** [comment]
@mpurzynski Just a quick question before merging, did you stage the removal of the original port scan alert?

**Comment by mpurzynski** [comment]
I did not, waiting for a review first.

File: alerts/nsm_scan_port.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        tags = ['nsm', "bro", 'portscan']
+
+        indicators = 'unknown'
+        #port = 'unknown'
+        # Maybe iterate through it?
+        x = aggreg['events'][0]['_source']
+        if 'details' in x:
+            if 'indicators' in x['details']:
+                indicators = x['details']['sourceipaddress']
+        #    if 'p' in x['details']:
```
[the remark this answers is not in the document]
Comment (mpurzynski) [reply alerts/nsm_scan_port.py]: Done, about to push that

File: tests/alerts/test_nsm_scan_port.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                "note": "Scan::Port_Scan",
+            }
+        }
+    }
+
+    # This alert is the expected result from running this task
+    default_alert = {
+        "category": "nsm",
+        "severity": "WARNING",
+        "summary": "Port scan from 10.99.88.77",
```
[the remark this answers is not in the document]
Comment (mpurzynski) [reply tests/alerts/test_nsm_scan_port.py]: This is an interesting challenge. We could sample ports or maybe even print the most popular ones? I need to do some further research and reach out to @JustinAzoff

File: alerts/nsm_scan_port.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# Copyright (c) 2018 Mozilla Corporation
+
+from lib.alerttask import AlertTask
+from query_models import SearchQuery, TermMatch, QueryStringMatch, PhraseMatch, ExistsMatch
```
Comment (pwnbus) [inline alerts/nsm_scan_port.py]: This needs to get updated to be as part of some of the work week changes:
```
from mozdef_util.query_models import SearchQuery, TermMatch, QueryStringMatch, PhraseMatch, ExistsMatch
```

File: tests/alerts/test_nsm_scan_port.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mpurzynski) [reply tests/alerts/test_nsm_scan_port.py]: Q1 - I need to rewrite Bro's port scanning detection for this.

**Pull request merged**